erts/emulator/beam/erl_db_catree.c

```diff
@@ -1196,7 +1196,7 @@ static void join_catree(DbTableCATree *tb,
     DbTableCATreeNode *neighbor_parent;
 
     ASSERT(thiz->is_base_node);
-    if (parent == NULL) {
+    if (parent == NULL || ERTS_IS_CRASH_DUMPING) {
```

Comment:
As I originally stated in #5981, it's possible to add such a condition. But it appears to be on a hot path, and it may affect performance to a degree.

Reply:
The random number generation is hot patch. But the split/join operations are not particularly hot as they only happen when a threshold of lock contention has been reached (split) or when it decides to randomly let "gravity" slowly shrink the tree (join). The split/join ops are quite heavy themself; seize locks, allocate/free memory, tree rotation.